File: src/devices/tests/v2/replace-failed-driver/driver.cc
// Copyright 2024 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <lib/driver/component/cpp/driver_base2.h>
#include <lib/driver/component/cpp/driver_export2.h>
#include <lib/driver/logging/cpp/logger.h>
#include <zircon/errors.h>

namespace {

class FailerDriver : public fdf::DriverBase2 {
 public:
  FailerDriver() : fdf::DriverBase2("failer") {}

  zx::result<> Start(fdf::DriverContext context) override {
    fdf::info("This driver is returning a ZX_ERR_NOT_SUPPORTED error.");
    return zx::error(ZX_ERR_NOT_SUPPORTED);
  }
};

}  // namespace

FUCHSIA_DRIVER_EXPORT2(FailerDriver);

File: src/ui/input/drivers/goodix-gt6853/rust/src/driver.rs
// Copyright 2026 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use fdf_component::{Driver, DriverContext, DriverError, Node, driver_register};
use log::info;

struct GoodixGt6853Driver {
    _node: Node,
}

driver_register!(GoodixGt6853Driver);

impl Driver for GoodixGt6853Driver {
    const NAME: &str = "goodix-gt6853";

    async fn start(mut context: DriverContext) -> Result<Self, DriverError> {
        info!("GoodixGt6853Driver (Rust Skeleton) started!");
        let _node = context.take_node()?;
        Ok(Self { _node })
    }

    async fn stop(&self) {
        info!("GoodixGt6853Driver (Rust Skeleton) stopped!");
    }
}

#[cfg(test)]
mod tests {
    #[test]
    fn test_placeholder() {}
}
